pallets/inflation/src/lib.rs

```diff
@@ -0,0 +1,638 @@
+            // The configuration & integration tests must ensure this never happens, so the following code is just an additional safety measure.
+            let blocks_per_cycle = match T::CycleConfiguration::blocks_per_cycle() {
+                0 => Balance::MAX,
+                blocks_per_cycle => Balance::from(blocks_per_cycle),
+            };
+
+            let build_and_earn_eras_per_cycle =
+                match T::CycleConfiguration::build_and_earn_eras_per_cycle() {
+                    0 => Balance::MAX,
+                    build_and_earn_eras_per_cycle => Balance::from(build_and_earn_eras_per_cycle),
+                };
+
+            let periods_per_cycle = match T::CycleConfiguration::periods_per_cycle() {
+                0 => Balance::MAX,
+                periods_per_cycle => Balance::from(periods_per_cycle),
+            };
+
+            // 3.1. Collator & Treausry rewards per block
+            let collator_reward_per_block = collators_emission / blocks_per_cycle;
+            let treasury_reward_per_block = treasury_emission / blocks_per_cycle;
+
+            // 3.2. dApp reward pool per era
+            let dapp_reward_pool_per_era = dapps_emission / build_and_earn_eras_per_cycle;
+
+            // 3.3. Staking reward pools per era
+            let base_staker_reward_pool_per_era =
+                base_stakers_emission / build_and_earn_eras_per_cycle;
+            let adjustable_staker_reward_pool_per_era =
+                adjustable_stakers_emission / build_and_earn_eras_per_cycle;
+
+            // 3.4. Bonus reward pool per period
+            let bonus_reward_pool_per_period = bonus_emission / periods_per_cycle;
+
+            // 4. Block at which the inflation must be recalculated.
+            let recalculation_block = now.saturating_add(T::CycleConfiguration::blocks_per_cycle());
+
+            // 5. Return calculated values
+            InflationConfiguration {
+                recalculation_block,
+                issuance_safety_cap,
+                collator_reward_per_block,
+                treasury_reward_per_block,
+                dapp_reward_pool_per_era,
+                base_staker_reward_pool_per_era,
+                adjustable_staker_reward_pool_per_era,
+                bonus_reward_pool_per_period,
+                ideal_staking_rate: params.ideal_staking_rate,
+            }
+        }
+
+        /// Check if payout cap limit would be reached after payout.
+        fn is_payout_cap_limit_exceeded(payout: Balance) -> bool {
+            let config = ActiveInflationConfig::<T>::get();
+            let total_issuance = T::Currency::total_issuance();
+
+            let new_issuance = total_issuance.saturating_add(payout);
+
+            if new_issuance > config.issuance_safety_cap {
+                log::error!("Issuance cap has been exceeded. Please report this issue ASAP!");
+            }
+
+            // Allow for 1% safety cap overflow, to prevent bad UX for users in case of rounding errors.
```

Comment:
We should create a follow-up ticket/issue to ensure this.

Reply:
The follow-up is to remove this completely 🙂 

This is just an initial safety measure to prevent accidental excessive minting.